mowgli_etl/pipeline/wdc/wdc_transformer.py

```diff
@@ -0,0 +1,134 @@
+                        (?: \d+)?\s?(?:x\s\d+\s?)?\w*", additional_info)
+
+            if len(dimensions) == 0:
+                dimensions = re.findall("\d+\s?\w+\s\d+\s?\w+\
+                        \slead\sx\s\d+\s?\w+", additional_info)
+
+        return dimensions
+
+    def transform(self, *, wdc_jsonl_file_path: Path)  -> Generator[Union[KgNode, KgEdge], None, None]:
+        # Prepare file and nlp
+        wdc_clean_file_path = self.clean(wdc_jsonl_file_path)
+        nlp = spacy.load("en_core_web_sm")
+        
+        # Parse file
+        with open(wdc_clean_file_path, mode="r") as data:
+            for row in data:
+                information = json.loads(row)
+                listing = information["title"]
+                description = information["description"]
+                additional_info = line["specTableContent"]
+                category = line["category"]
+
+                # Ignore product if there is no listing name
+                if listing == None:
+                    continue
+
+                doc = nlp(listing)
+
+                last_noun_name = ""
+                first_noun_sequence_name = ""
+                first_noun_flag = 0
+                last_noun_sequence_name = ""
+                last_noun_flag = 1
+
+                for token in doc:
+                    if token.pos in range(92, 101):
+                        # Assume that general product name is last noun in title
+                        last_noun_name = token.text
+
+                        # Assume that general product name is the first sequence of just nouns
+                        if first_noun_flag == 0:
+                            if first_noun_sequence_name != "":
+                                first_noun_sequence_name += " "
+                            first_noun_sequence_name += token.text
+
+                        # Assume that general product name is the last sequence of just nouns
+                        if last_noun_flag == 1:
+                            last_noun_sequence_name = ""
+                            last_noun_flag = 0
+                        if last_noun_sequence_name != "":
+                            last_noun_sequence_name += " "
+                        last_noun_sequence_name += token.text
+
+                    else:
+                        # Throw flag to terminate first noun sequence
+                        if first_noun_sequence_name != "":
+                            first_noun_flag = 1
+                        last_noun_flag = 1
+
+                first_noun_sequence_name.rstrip(" ")
+
+                dimensions = self.__find_dimensions(
+                        description,
+                        listing, 
+                        additional_info)
+
+                general_name = f"{last_noun_name} or\
+                        {first_noun_sequence_name} or\
+                        {last_noun_sequence_name}"
+
+                yield KgNode(id = f"{WDC_DATASOURCE_ID}:\"general_name\"",
```

Comment:
This is fine as a start. You can yield a KgEdge that connects IsA to /c/en/product.